[mlir][gpu] Improve `gpu-lower-to-nvvm-pipeline` Documentation

grypp opened the pull request:
This PR improves the documentation for the `gpu-lower-to-nvvm-pipeline` (as it was remaning item for #75775)

- Changes pipeline `gpu-lower-to-nvvm` -> `gpu-lower-to-nvvm-pipeline`
- Adds a section in GPU Dialect in website. It clarifies the pipeline's functionality in lowering primary dialects to NVVM targets.

grypp committed:
[mlir][gpu] Improve `gpu-lower-to-nvvm-pipeline` Documentation

This PR improves the documentation for the `gpu-lower-to-nvvm-pipeline` (as it was remaning item for #75775)

- Changes pipeline `gpu-lower-to-nvvm` -> `gpu-lower-to-nvvm-pipeline`
- Adds a section in GPU Dialect in website. It clarifies the pipeline's functionality in lowering primary dialects to NVVM targets.

grypp commented:
I continued to provide support for 'linalg' with 'ConvertLinalgToLoops.' I mentioned [here](https://github.com/grypp/llvm-project/blob/nvgpu-compiler/mlir/docs/Dialects/GPU.md?plain=1#L67), the pipeline requires explicitly parallel IR and doesn't do parallelism to any of the operations especially for `linalg`. Does this explanation align with our goals?

joker-eph commented:
> The pipeline requires explicitly parallel IR and doesn't do parallelism to any of the operations especially for linalg. Does this explanation align with our goals?

I would think that "gpu" is already parallel code (the pipeline is "gpu-to-nvvm" here): so it seems to me that handling of linalg should be done outside of this pipeline  and this pipeline invoked on the results.

joker-eph commented on mlir/docs/Dialects/GPU.md:
```diff
@@ -60,6 +60,38 @@ mlir-translate example-nvvm.mlir        \
+
+This pipeline specifically requires explicitly parallel IR and doesn't do GPU parallelization. For example, it supports linalg but it gets translated into sequential loops. To incorporate parallelism, necessary transformations must be applied before utilizing this pipeline.
+
+It's designed to provide a generic solution for NVVM targets, generating NVVM and LLVM dialect code compatible with `mlir-cpu-runner` or execution engine.
```
Can you wrap this? `mdformat -wrap 80` should do.

grypp committed:
remove linalg
run mdformat --wrap 80 on markdown

grypp commented:
> > The pipeline requires explicitly parallel IR and doesn't do parallelism to any of the operations especially for linalg. Does this explanation align with our goals?
> 
> I would think that "gpu" is already parallel code (the pipeline is "gpu-to-nvvm" here): so it seems to me that handling of linalg should be done outside of this pipeline and this pipeline invoked on the results.

I see yor point, it makes sense. I removed the lowering of linalg. In integration tests I added `convert-linalg-to-loops` before `gpu-to-nvvm-pipeline`.

joker-eph reviewed (approved):
Thanks!
